python/ray/data/tests/test_dataset.py

```diff
@@ -1739,6 +1739,25 @@ def test_add_column(ray_start_regular_shared):
         ds = ray.data.range(5).add_column("value", 0)
 
 
+def test_drop_columns(ray_start_regular_shared, tmp_path):
+    df = pd.DataFrame({"col1": [1, 2, 3], "col2": [2, 3, 4], "col3": [3, 4, 5]})
+    ds1 = ray.data.from_pandas(df)
+    ds1.write_parquet(str(tmp_path))
+    ds2 = ray.data.read_parquet(str(tmp_path))
+
+    for ds in [ds1, ds2]:
+        assert ds.drop_columns(["col2"]).take(1) == [{"col1": 1, "col3": 3}]
+        assert ds.drop_columns(["col1", "col3"]).take(1) == [{"col2": 2}]
+        assert ds.drop_columns([]).take(1) == [{"col1": 1, "col2": 2, "col3": 3}]
+        assert ds.drop_columns(["col1", "col2", "col3"]).take(1) == [{}]
+        assert ds.drop_columns(["col1", "col1", "col2", "col1"]).take(1) == [
+            {"col3": 3}
+        ]
+        # Test dropping non-existent column
+        with pytest.raises(KeyError):
```

Comment:
Here is an example of exception stack trace:

```
python/ray/data/tests/test_dataset.py::test_drop_columns (_map_block_nosplit pid=22716) 2022-06-29 12:14:44,999 INFO worker.py:749 -- Task failed with retryable exception: TaskID(8f73ff1cc11681cdffffffffffffffffffffffff01000000).
(_map_block_nosplit pid=22716) Traceback (most recent call last):
(_map_block_nosplit pid=22716)   File "python/ray/_raylet.pyx", line 670, in ray._raylet.execute_task
(_map_block_nosplit pid=22716)     with ray._private.worker._changeproctitle(title, next_title):
(_map_block_nosplit pid=22716)   File "python/ray/_raylet.pyx", line 674, in ray._raylet.execute_task
(_map_block_nosplit pid=22716)     outputs = function_executor(*args, **kwargs)
(_map_block_nosplit pid=22716)   File "/Users/chengsu/ray/python/ray/data/_internal/compute.py", line 437, in _map_block_nosplit
(_map_block_nosplit pid=22716)     for new_block in block_fn(block, *fn_args, **fn_kwargs):
(_map_block_nosplit pid=22716)   File "/Users/chengsu/ray/python/ray/data/dataset.py", line 441, in transform
(_map_block_nosplit pid=22716)     applied = batch_fn(view, *fn_args, **fn_kwargs)
(_map_block_nosplit pid=22716)   File "/Users/chengsu/ray/python/ray/data/dataset.py", line 561, in <lambda>
(_map_block_nosplit pid=22716)     lambda batch: batch.drop(columns=cols), compute=compute, **ray_remote_args
(_map_block_nosplit pid=22716)   File "/opt/homebrew/lib/python3.9/site-packages/pandas/util/_decorators.py", line 311, in wrapper
(_map_block_nosplit pid=22716)     return func(*args, **kwargs)
(_map_block_nosplit pid=22716)   File "/opt/homebrew/lib/python3.9/site-packages/pandas/core/frame.py", line 4954, in drop
(_map_block_nosplit pid=22716)     return super().drop(
(_map_block_nosplit pid=22716)   File "/opt/homebrew/lib/python3.9/site-packages/pandas/core/generic.py", line 4267, in drop
(_map_block_nosplit pid=22716)     obj = obj._drop_axis(labels, axis, level=level, errors=errors)
(_map_block_nosplit pid=22716)   File "/opt/homebrew/lib/python3.9/site-packages/pandas/core/generic.py", line 4311, in _drop_axis
(_map_block_nosplit pid=22716)     new_axis = axis.drop(labels, errors=errors)
(_map_block_nosplit pid=22716)   File "/opt/homebrew/lib/python3.9/site-packages/pandas/core/indexes/base.py", line 6644, in drop
(_map_block_nosplit pid=22716)     raise KeyError(f"{list(labels[mask])} not found in axis")
(_map_block_nosplit pid=22716) KeyError: "['dummy_col'] not found in axis"
```

Not sure if the stack trace is intuitive enough for our users when they accidentally drop non-existent column. Please let me know if we want to catch the exception and add more description.